tests/compile-test.rs

```diff
@@ -168,7 +168,13 @@ fn run_ui() {
     let _threads = VarGuard::set(
         "RUST_TEST_THREADS",
         // if RUST_TEST_THREADS is set, adhere to it, otherwise override it
-        env::var("RUST_TEST_THREADS").unwrap_or_else(|_| num_cpus::get().to_string()),
+        env::var("RUST_TEST_THREADS").unwrap_or_else(|_| {
+            match std::thread::available_parallelism() {
+                Ok(threads) => threads.get(),
+                _ => 1,
+            }
```

Comment:
Wouldn't that be the same as

```suggestion
            std::thread::available_parallelism().map_or(1, |t| t.get())
```

Reply:
To `map_or` not to `map`, that is the question!

Yeah, should be identical, thanks :D

Reply:
Also apparently we don't have a lint for that yet :P

Reply:
https://github.com/rust-lang/rust-clippy/issues/8492